pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -228,8 +255,69 @@ func (r *ReconcileIBMBlockCSI) Reconcile(request reconcile.Request) (reconcile.R
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileIBMBlockCSI) addFinalizerIfNotPresent(instance *ibmblockcsi.IBMBlockCSI) error {
+	logger := log.WithName("addFinalizer")
+
+	accessor, finalizerName, err := r.getAccessorAndFinalizerName(instance)
+	if err != nil {
+		return err
+	}
+
+	if !util.Contains(accessor.GetFinalizers(), finalizerName) {
+		logger.Info("adding", "finalizer", finalizerName, "on", accessor.GetName())
+		accessor.SetFinalizers(append(accessor.GetFinalizers(), finalizerName))
+
+		if err := r.client.Update(context.TODO(), instance.Unwrap()); err != nil {
```

Comment:
Same as in https://github.com/IBM/ibm-block-csi-operator/pull/133/files#r584316456